content/docs/04-clusters/03-edge/04-edgeforge-workflow/03.5-palette-canvos.md

```diff
@@ -0,0 +1,658 @@
+
+
+10. CanvOS utility internally uses [Earthly](https://earthly.dev/) to build the target artifacts. Issue the following command to invoke the Earthly build. The `--PE_VERSION` option in the command below signifies the Palette Edge version to use, which refers to the git tag you checked out in the previous step. 
+<br />
+  <WarningBox>
+
+  Ensure to issue the command below with root-level or `sudo` privileges.
+
+  </WarningBox>
+
+  ```shell
+  ./earthly.sh +build-all-images --PE_VERSION=$(git describe --abbrev=0 --tags)
+  ```
+
+  ```bash coloredLines=2-2
+  # Output condensed for readability
+  ===================== Earthly Build SUCCESS ===================== 
+  Share your logs with an Earthly account (experimental)! Register for one at https://ci.earthly.dev.
+  ```
+  This command may take up to 15-20 minutes to finish depending on the resources of the host machine.
+  <br />
+
+
+11. List the Docker images to verify the two provider OS images - one compatible with K3s v1.24.6 and another with K3s v1.25.2.
+<br />
+
+  ```shell
+  docker images
+  ```
+
+  ```bash coloredLines=3-4
+  # Output
+  REPOSITORY        TAG                                IMAGE ID       CREATED         SIZE
+  ttl.sh/ubuntu     k3s-1.25.2-v3.4.3-palette-learn    b3c4956ccc0a   6 minutes ago   2.49GB
+  ttl.sh/ubuntu     k3s-1.24.6-v3.4.3-palette-learn    fe1486da25df   6 minutes ago   2.49GB
+  earthly/earthly   v0.7.4                             d771cc8edc38   2 weeks ago     333MB
+  ```
+  <br />
+
+
+12. To use the provider OS images in your cluster profile, push them to the image registry mentioned in the **.arg** file. The current example and default behavior uses the [ttl.sh](https://ttl.sh/) image registry. This image registry is free to use and does not require a sign-up. Images pushed to *ttl.sh* are ephemeral and will expire after the 24 hrs time limit.  Use the following commands to push the provider OS image to the *ttl.sh* image registry.  
+  <br />
+
+  ```bash
+  docker push ttl.sh/ubuntu:k3s-1.25.2-v3.4.3-palette-learn
+  docker push ttl.sh/ubuntu:k3s-1.24.6-v3.4.3-palette-learn
+  ```
+  <br />
+
+  <WarningBox>
+
+  As a reminder, [ttl.sh](https://ttl.sh/) is a short-lived image registry. If you do not use these provider OS images in your cluster profile within 24 hours of pushing to *ttl.sh*, they will expire and must be re-pushed. Refer to the Advanced workflow in the current guide to learn how to use another registry, such as Docker Hub, and tag the docker images accordingly.
+
+  </WarningBox>
+  <br />
+
+
+13. After pushing the provider images to the image registry, add the following custom content within the `options` attribute of your cluster profile's **OS layer** so that the cluster profile can pull the provider OS image from the image registry. 
+
+  If you used customized arguments and Dockerfile to build the Edge artifacts, change these attributes' values, as applicable to you, before using them in your cluster profile.
```

Comment:
We should mention that the OS pack should be the BYOOS pack.  We should also mention that the `system.XXXX` values come from what is used in the `.arg` file.